AirCasting/Notes/EditNote/EditNoteView.swift

```diff
@@ -0,0 +1,83 @@
+// Created by Lunar on 16/12/2021.
+//
+import SwiftUI
+import AirCastingStyling
+
+struct EditNoteView<VM: EditNoteViewModel>: View {
+    @StateObject var viewModel: VM
+    @Binding var editNoteModal: Bool
+    
+    var body: some View {
+        VStack(alignment: .leading, spacing: 20) {
+            title
+            description
+            noteField
+            continueButton
+            deleteButton
+            cancelButton
+        }.padding()
+    }
+}
+
+private extension EditNoteView {
+    
+    var title: some View {
+        Text(Strings.EditNoteView.title)
+            .font(Fonts.boldTitle4)
+            .foregroundColor(.darkBlue)
+    }
+    
+    var description: some View {
+        Text(Strings.EditNoteView.description)
+            .font(Fonts.muliHeading2)
+            .foregroundColor(.aircastingGray)
+    }
+    
+    var noteField: some View {
+        TextField(Strings.EditNoteView.placeholder, text: $viewModel.noteText)
+            .padding()
+            .frame(minWidth: UIScreen.main.bounds.width - 40,
+                   maxWidth: UIScreen.main.bounds.width - 40,
+                   minHeight: (UIScreen.main.bounds.height) / 3 < 200 ? (UIScreen.main.bounds.height / 3) : 200,
+                   maxHeight: 200,
+                   alignment: .topLeading)
+            .background(Color.aircastingGray.opacity(0.05))
+            .border(Color.aircastingGray.opacity(0.1))
+    }
+    var continueButton: some View {
+        Button {
+            //)
+        } label: {
+            Text(Strings.EditNoteView.saveButton)
+                .bold()
+        }
+        .buttonStyle(BlueButtonStyle())
+    }
+    
+    var deleteButton: some View {
+        Button {
+            //
+        } label: {
+            Text(Strings.EditNoteView.deleteButton)
+                .bold()
+        }
+        .buttonStyle(BlueButtonStyle())
+    }
+    
+    var cancelButton: some View {
+        Button {
+            editNoteModal.toggle()
```

Comment:
I feel like this too should be handled by the viewmodel

Reply:
As long as in this and above I used Binding it is not easy for me to do. Is it worth doing in the case of Binding? If so I will dig deeper

Reply:
It can be done using routing closures (so that whoever creates this scene needs to provide view routings for certain scenarios). I will create a PR to this PR so you can review and see how I'd do it.

Reply:
Done. see #528

Reply:
Thank you @pawelgil. Learning it and looks awesome